server/driver/wivrn_session.cpp

```diff
@@ -418,8 +424,26 @@ void wivrn_session::operator()(from_headset::trackings && tracking)
 		        std::string(magic_enum::enum_name(roles.right_profile)).c_str(),
 		        std::string(magic_enum::enum_name(right)).c_str());
 		std::lock_guard lock(roles_mutex);
+
+		// don't change role when hand from other driver is used
+		if (roles.left == left_hand_interaction_index || roles.left == left_controller_index)
+		{
+			if (left == XRT_DEVICE_EXT_HAND_INTERACTION)
+				roles.left = left_hand_interaction_index;
+			else
+				roles.left = left_controller_index;
```

Comment:
We may have to refresh the xxx_list objects in the controller/hand_interaction so that the `last_request` won't trigger a deactivation